.github/workflows/sync-to-upstream.yml

```
@@ -78,21 +67,20 @@ jobs:
     name: Sync to Upstream Repository
     runs-on: ubuntu-latest
     
-    # Only run when PR is actually merged (not just closed)
-    # Branch check is done at step level where env is available
-    if: github.event.pull_request.merged == true
+    # Only run when PR is merged and running in fork repository
+    if: |
+      github.event.pull_request.merged == true &&
```

actionlint (1.7.8)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)